.circleci/config.yml

```diff
@@ -0,0 +1,109 @@
+version: 2
+jobs:
+  build-font:
+    working_directory: ~/circleci-code-solidaire
+
+    docker:
+          - image: circleci/node:10-browsers
+    steps:
+      - checkout
+
+      - restore_cache:
+          key: yarn-
+
+      - run:
+          name: Download dependencies (frontend)
+          working_directory: frontend
+          command: yarn install
+
+      - run:
+           name: Front tests
+           working_directory: frontend
+           command: yarn test --watch=false --progress=false
+
+      - run:
+           name: Build frontend
+           working_directory: frontend
+           command: yarn build --prod
+
+      - persist_to_workspace:
+           root: frontend
+           paths:
+             - dist
+
+      - save_cache:
+           key: yarn-{{ checksum "frontend/package.json" }}
+           paths:
+             - ~/.cache/yarn
+             - frontend/node_modules
+             
+  build-back:
+    working_directory: ~/circleci-code-solidaire
+
+    docker:
+      - image: circleci/openjdk:8-jdk-browsers
+    steps:
+
+      - checkout
+
+      - setup_remote_docker
+
+      - attach_workspace:
+          at: frontend
+
+      - run: ls -al frontend/dist
+
+      - run: mkdir -p backend/target/classes/static/
+
+      - run: cp -a frontend/dist/frontend/. backend/target/classes/static/
+
+      - run: ls -al backend/target/classes/static/
+
+      - restore_cache:
+          key: maven-
+
+      - run:
+          name: Download dependencies (backend)
+          working_directory: backend
+          command: mvn dependency:go-offline
+
+      - run:
+         name: Build application
+         command: mvn package --quiet
+                           
+      - save_cache:
+         key: maven-{{ checksum "pom.xml" }}-{{checksum "backend/pom.xml"}}-{{checksum "frontend/pom.xml"}}
+         paths:
+           - ~/.m2
+
+      - run:
+          name: Build Docker image (backend)
+          working_directory: backend
+          command: mvn package --quiet -DskipTests && docker build -t backend .
+  deploy:
+    working_directory: ~/circleci-code-solidaire
+    machine:
+      enabled: true
+    steps:
+      - checkout
+      - run:
+          name: Deploy Master to Heroku
+          working_directory: backend
+          command: docker login --username=abid1991@hotmail.fr --password=eb6e5e1e-d611-4ecb-9921-4d2b0c1ccc6d registry.heroku.com
```

Comment:
:scream: Use environment variables in circleci !!! not plain text password :sweat:

Reply:
Yeah. I want first to make it work . Next I will hide it.
I have a problem when I push the image to heroku registery 😅